build_docs.py

```diff
@@ -34,19 +36,18 @@
 import shutil
 import subprocess
 import sys
-import time
 from bisect import bisect_left as bisect
 from collections import OrderedDict
-from contextlib import contextmanager
 from pathlib import Path
 from string import Template
 from textwrap import indent
+from typing import Iterable
 
 import zc.lockfile
 import jinja2
 import requests
+import tomlkit
```

Comment:
Use tomllib for forward-compatibility?
```suggestion
try:
    import tomllib
except ImportError:
    import tomli as tomllib 
```

Reply:
Not sure as long as the stdlib tomllib can't write.

For the next PR I'm experimenting with using `toml` to store the state of the builds, like:

```toml
["/uk/3.12/"]
cpython_sha = "0102faf0cac107fd4d0fde9902560c7cf0a3be9d"
translation_sha = "3e7604d344c4191b8f452a61015c9e34ba8b5f09"
last_build = 2023-10-13T20:03:37.732662Z
last_build_duration = 103.98882411496015

["/tr/3.12/"]
cpython_sha = "0102faf0cac107fd4d0fde9902560c7cf0a3be9d"
translation_sha = "cefb12aeefe969b0227a599024c6cce708965982"
last_build = 2023-10-13T20:05:14.523090Z
last_build_duration = 96.78385900298599
```

nothing manually involved, just the script keeping its state up-to-date in the file in order to avoid rebuilding docs that has not changed.

TOML is probably not intended to be used as a database, but for like 2k bytes it's probably more human friendly than sqlite or pickle.

Reply:
@JulienPalard can you create a draft PR after this one? I would like to see how it evolves.

Reply:
Here you go: https://github.com/python/docsbuild-scripts/pull/171

But better read it locally, maybe, as it's based on top of the `update` branch.

Reply:
Thanks!